config/workflow/environment.ts

```diff
@@ -4,22 +4,38 @@ import * as Fs from 'fs';
 import * as Path from 'path';
 import { merge } from 'lol/js/object';
 import template from "lodash.template";
+import dotenv from "dotenv";
+
 
 export function loadYML(path: string) {
   try {
     let content = Fs.readFileSync(path, 'utf-8')
+    let temp = template(content);
+    content = temp({ ENV: process.env });
     return YAML.safeLoad(content) || {}
-  } catch(e) {
+  } catch (e) {
     throw `Missing environment file "${path}"`
   }
 }
 
+
+
 export function parse(config: WKEnv = { environment: 'development' }, callback?: (config: WKEnv) => void) {
+
+
+  let dotenvpath = Path.resolve(process.cwd(), '.env.' + config.environment);
```

Comment:
I looked `dotenv` library, it does nothing more that adding variables to process.env. I do not think we need more dependency to install, we can use `js-yaml` to do the same work.

Process:
* Create your `.env.${ENVIRONMENT}` at root
* Use loadYML(path) returns a JSON
* `Object.assign(process.env, JSON)` or you can escape override like dotenv [here](https://github.com/motdotla/dotenv/blob/master/lib/main.js#L98)

Your `.env.${ENVIRONMENT}` looks like that

```yaml
AIRTABLE_APIKEY: "12345"
AIRTABLE_APPID: ABCDE
AIRTABLE_TAB: SOME_TAB
AIRTABLE_VIEW: SOME_VIEW
```

Reply:
It's more future proof if we want to add some tag directives like [wk](https://github.com/wk-js/command/blob/v2/tests/Commands.yml)

Reply:
I got the point that dotenv only adds variables to process.env and maybe we can get rid of this dependency. But `.env` files ( with the syntax `variable=value` and not `variable: value` ) are a common way to inject environment variables to any kind of techno. 
So env variables can be shared between backend and frontend code, or any sub system inside the frontend website. And the definition follow an established convention.
We can maybe embed a custom parser ( wich can be very lightweight ), to keep the usual way of writing .env files ?

Reply:
I understand. It can be a parse solution:

```js
const lines = file.split("\n")
lines.forEach(line => {
  const kv = line.split("=")
  process.env[kv[0]] = kv[1]
})
```

Reply:
I will investigate this sllution. Thanks !